[python-updates] python3Packages.gradio-client: 0.5.0 -> 0.7.3

pbsds opened the pull request:
## Description of changes

context: https://github.com/NixOS/nixpkgs/pull/273538#issuecomment-1856895396

~~Can't test due to an [eval error](https://gist.github.com/pbsds/064ee4de502f16a3ddfb489e48cdbcea).~~

@mweinelt: do you prefer PRs or is it more normal/preferred to push directly to `python-updates`?


<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage committed:
python311Packages.transformers: 4.35.2 -> 4.36.1

Changelog: https://github.com/huggingface/transformers/releases/tag/v4.36.1

pbsds committed:
python3Packages.gradio-client: 0.5.0 -> 0.7.3

pbsds committed:
python3Packages.gradio: 4.7.1 -> 4.9.1

mweinelt commented:
> google-cloud-cpp = disable-warnings-if-gcc13 (callPackage ../development/libraries/google-cloud-cpp { });

This is due to `disable-warninigs-if-gcc13` settings `env.NIX_CFLAGS_COMPILE`, as well as the derivation itself. https://github.com/NixOS/nixpkgs/pull/268097#issuecomment-1848776664

> do you prefer PRs or is it more normal/preferred to push directly to python-updates?

If you are confident about the change, you can push it there.

pbsds committed:
intel-media-sdk: fix gcc13 build with patch

pbsds committed:
python3Packages.psygnal: add missing checkPhase input

pbsds committed:
python3Packages.zarr: propagate msgpack

pbsds committed:
python3Packages.numcodecs: add msgpack to checkPhase

Increases test coverage of functionality used downstream in zarr

pbsds committed:
google-cloud-cpp: fix build with patch from upstream

The patch is merged in 2.16.0, but bumping is not
trivial and my machine takes 1h12m57s to build this.

pbsds committed:
fixup! python311Packages.pyarrow: relax setuptools_scm constraint

pbsds committed:
python3Packages.tokenizers: 0.14.1 -> 0.15.0

Diff: https://github.com/huggingface/tokenizers/compare/v0.14.1...v0.15.0

pbsds committed:
fixup! python3Packages.scikit-image: 0.21.0 -> 0.22.0

pbsds committed:
python3Packages.gradio-pdf: init at 0.0.3

pbsds committed:
python3Packages.datasets: remove pyarrow<14.0.1 vulnerability fix

pbsds commented:
@mweinelt gradio now builds with all the fixes added. Feel free to merge or rebase as you see fit

I ended up cherry-picking a bump from #273538, i assume it won't cause a merge conflict
